### PR TITLE
Merge layernorm trt fuse

### PR DESCRIPTION
**Pull request by wwbitejotunn**
### PR types
Performance optimization

### PR changes
OPs

### Describe
Fuse patch merge and layernorm in trt
For swin tiny model, it speed up from 1.14 ms to 1.06 ms in A10

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/plugin/merge_layernorm_op_plugin.cu [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+__inline__ __device__ T warpReduceSum(T val) {
+#pragma unroll
+  for (int mask = 16; mask > 0; mask >>= 1)
+    val += __shfl_xor_sync(FINAL_MASK, val, mask, 32);
+  return val;
+}
+
+/* Calculate the sum of all elements in a block */
+template <typename T>
+__inline__ __device__ T blockReduceSum(T val) {
```
Comment (zhangjun) [inline paddle/fluid/inference/tensorrt/plugin/merge_layernorm_op_plugin.cu]: 请复用phi算子库

Comment (wwbitejotunn) [reply paddle/fluid/inference/tensorrt/plugin/merge_layernorm_op_plugin.cu]: reduceSum已复用phi算子库

**Review by b3602sss — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_merge_layernorm_fuse_pass PROPERTIES TIMEOUT 180)`

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**